File: markdowner-core/src/test/java/org/jesperancinha/parser/markdowner/filter/NPMFilterTest.java
package org.jesperancinha.parser.markdowner.filter;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

import java.nio.file.Path;


public class NPMFilterTest {

    @Test
    public void testNPMFilter() {
        final Path resource = Path.of(getClass().getResource("/directory2NoReadme/project3MavenAndNPM/package.json").getPath());
        final ProjectFilter<Path> npmFilter = new NPMFilter();

        boolean result = npmFilter.test(resource);

        assertThat(result).isTrue();
        assertThat(npmFilter.lastProjectName()).isEqualTo("npm-project");
    }

    @Test
    public void testNPMFilterFail() {
        final Path resource = Path.of(getClass().getResource("/directory2NoReadme/project1Maven/pom.xml").getPath());
        final ProjectFilter<Path> mavenFilter = new NPMFilter();

        boolean result = mavenFilter.test(resource);

        assertThat(result).isFalse();
    }
}

File: markdowner-filter/src/test/kotlin/org/jesperancinha/parser/markdowner/filter/NPMFilterTest.kt
package org.jesperancinha.parser.markdowner.filter

import io.kotest.matchers.booleans.shouldBeFalse
import io.kotest.matchers.booleans.shouldBeTrue
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import java.nio.file.Path

class NPMFilterTest {
    @Test
    fun testNPMFilter() {
        val packageJson = javaClass.getResource("/directory2NoReadme/project3MavenAndNPM/package.json")
        packageJson.shouldNotBeNull()
        val resource = Path.of(packageJson.path)
        val npmFilter: ProjectFilter<Path> = NPMFilter()
        val result = npmFilter.test(resource)
        result.shouldBeTrue()
        npmFilter.lastProjectName() shouldBe "npm-project"
    }

    @Test
    fun testNPMFilterFail() {
        val pomXml = javaClass.getResource("/directory2NoReadme/project1Maven/pom.xml")
        pomXml.shouldNotBeNull()
        val resource = Path.of(pomXml.path)
        val mavenFilter: ProjectFilter<Path> = NPMFilter()
        val result = mavenFilter.test(resource)
        result.shouldBeFalse()
    }
}